index/server/Dockerfile

```diff
@@ -1,19 +1,34 @@
-# Builder image
-FROM golang:alpine3.11 AS builder
+# Index Server build stage
+FROM golang:alpine3.11 AS index-builder
 WORKDIR /tools
 COPY . .
 RUN CGO_ENABLED=0 go build -mod=vendor -o index-server main.go
 
+# Registry Viewer build stage
+FROM registry.access.redhat.com/ubi8/nodejs-14-minimal AS node-builder
+USER root
+WORKDIR /app
+RUN npm install -g yarn
+COPY registry-viewer . 
+
+ENV DEVFILE_VIEWER_ROOT /viewer
+ENV DEVFILE_COMMUNITY_HOST false
+
+# Build the devfile registry viewer
+RUN $(npm get prefix)/bin/yarn install --frozen-lockfile --ignore-optional
+RUN $(npm get prefix)/bin/yarn build
+RUN $(npm get prefix)/bin/yarn install --production --ignore-scripts --prefer-offline --ignore-optional
```

Comment:
any idea how we can fetch the dockerfile steps from registry-viewer rather than having two copies of the same logic 🤔 (not that we need to do it now)

Reply:
You could do two separate docker builds:
1) Build the registry viewer image, using its dockerfile (build stage only, so need to specify that as a flag)
2) Build the index server. During the runtime stage, copy over the viewer resources from the viewer container built in step 1

There isn’t really a way to merge the two runtime stages (the index server and registry viewer) into one without doing something really messy, so I’d keep that separate

Reply:
I’ll see how feasible this is, if it’s nothing quick, I’ll shelve it for later

Reply:
Fixed.